File: adept-core/src/main/scala/adept/artifact/models/ArtifactAttribute.scala
package adept.artifact.models

import adept.utils.OrderingHelpers

case class ArtifactAttribute(name: String, values: Set[String])

object ArtifactAttribute {
  implicit val ordering: Ordering[ArtifactAttribute] = new Ordering[ArtifactAttribute] {
    def compare(x: ArtifactAttribute, y: ArtifactAttribute): Int = {
      if (x.name < y.name)
        -1
      else if (x.name > y.name)
        1
      else {
        assert(x.name == y.name)
        OrderingHelpers.stringSetCompare(x.values, y.values)
      }
    }
  }
}

File: adept-lockfile/src/main/java/adept/lockfile/ArtifactAttribute.java
package adept.lockfile;

import java.util.Set;

class ArtifactAttribute {
  final String name;
  final Set<String> values;
  
  ArtifactAttribute(String name, Set<String> values) {
    this.name = name;
    this.values = values;
  }
}
